enclave/sgx/collateralinfo.c

```diff
@@ -169,6 +208,133 @@ oe_result_t oe_get_sgx_quote_verification_collateral(
     return result;
 }
 
+void oe_prealloc_quote_verification_collateral_args(
+oe_get_sgx_quote_verification_collateral_args_t* buf, oe_get_sgx_quote_verification_collateral_args_t* default_sizes)
+{
+    /**These numbers are size estimates of various
+     * quote_verification_collateral_args that
+     * are passed into oe_get_quote_verification_
+     * collateral_ocall().
+     * */
+    default_sizes->tcb_info_size = TCBINFO_DEFAULT_SIZE;
+    default_sizes->tcb_info_issuer_chain_size = ALL_ISSUER_CHAIN_DEFAULT_SIZE;
+    default_sizes->pck_crl_size = PCK_CRL_DEFAULT_SIZE;
+    default_sizes->root_ca_crl_size = ROOT_CA_CRL_DEFAULT_SIZE;
+    default_sizes->pck_crl_issuer_chain_size = ALL_ISSUER_CHAIN_DEFAULT_SIZE;
+    default_sizes->qe_identity_issuer_chain_size = ALL_ISSUER_CHAIN_DEFAULT_SIZE;
+    default_sizes->qe_identity_size = QE_IDENTITY_DEFAULT_SIZE;
```

Comment:
This will always overwrite the cached values whenever this function is called. You can probably set these when initializing the struct in order to ensure they don't get updated on every call.